### DISCUSSION TITLE
A JavaScript error occurred in the main process

### DISCUSSION THREAD

**Discussion by ghost** [agalwood/Motrix#1373] [Q&A]
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module 'fs/promises'
Require stack:
- /app/motrix/resources/app.asar/node_modules/electron-updater/out/AppUpdater.js
- /app/motrix/resources/app.asar/node_modules/electron-updater/out/main.js
- /app/motrix/resources/app.asar/dist/electron/main.js
- 
    at Module._resolveFilename (internal/modules/cjs/loader.js:972:15)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:249:921)
    at Module._load (internal/modules/cjs/loader.js:848:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12633)
    at Module.require (internal/modules/cjs/loader.js:1032:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/app/motrix/resources/app.asar/node_modules/electron-updater/out/AppUpdater.js:8:20)
    at Module._compile (internal/modules/cjs/loader.js:1152:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1173:10)
    at Module.load (internal/modules/cjs/loader.js:992:32)
Gtk-Message: 18:34:51.508: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:51.508: Failed to load module "pk-gtk-module"
Gtk-Message: 18:34:51.509: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:51.509: Failed to load module "pk-gtk-module"

**Comment by ghost** [comment]
flapak 1.6.11

**Comment by proofmancer** [comment]
This error message is indicating that the module 'fs/promises' cannot be found. This error is often caused by using a version of Node.js that doesn't support ES6 module syntax, which includes the 'fs/promises' module. The module was introduced in **Node.js version 14.**

Here's what you can do to address this issue:

1. **Check Node.js Version:** Make sure you're using a version of Node.js that supports ES6 modules. Update your Node.js version to at least 14 if you're not already using it.

2. **Check Electron Version:** If you're using Electron, make sure you're using a version that supports ES6 modules. Starting from Electron 14, you can use ES6 modules in your Electron app.

3. **Check Your Code:** Ensure that you're using the correct syntax to import modules. For example, if you're using ES6 modules, you should use the import syntax instead of the require syntax:
```
// Correct ES6 module import
import fs from 'fs/promises';
```
If you're using CommonJS modules (Node.js versions prior to 14), you should use the require syntax:
```
// Correct CommonJS module import
const fs = require('fs').promises;

```

4. **Check Dependencies:** If your application or any of its dependencies use the 'fs/promises' module, ensure that the versions of these dependencies are compatible with your Node.js and Electron versions.
